### PR TITLE
chore: Add API method DeleteAccount

### PR DESCRIPTION
**Pull request by jefft0**
Add DeleteAccount to the gRPC API and add deleteAccount to useGno().

### CODE REVIEW

**Review by D4ryl00 — APPROVED** [review]
LGTM

File: service/rpc/rpc.proto [hunk 1]
```diff
@@ -41,6 +41,10 @@ service GnomobileService {
   // length of the result.)
   rpc GetActiveAccount(GetActiveAccountRequest) returns (GetActiveAccountResponse);
 
+  // DeleteAccount deletes the account with the given name, using the password to
+  // ensure access. If the account doesn't exist, then return ErrCryptoKeyNotFound.
```
[the remark this answers is not in the document]
Comment (D4ryl00) [reply service/rpc/rpc.proto:45]: If you agree, there is also another location to update this comment.

[the remark this answers is not in the document]
Comment (jefft0) [reply service/rpc/rpc.proto:45]: Good point. But maybe a bad password should be a different error. The Go code returns "ciphertext decryption failed". Do you think we should add ErrDecryptionFailed to the RPC ErrCode?

**Review by iuricmp — APPROVED** [review]
👍

File: service/rpc/rpc.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (jefft0) [reply service/rpc/rpc.proto:45]: Hi @D4ryl00 . I did a force push to add the error code ErrDecryptionFailed and to update the doc comment. Does it look good?

File: service/api.go [hunk 2]
```diff
@@ -155,6 +156,22 @@ func (s *gnomobileService) GetActiveAccount(ctx context.Context, req *connect.Re
...
 }
 
+// DeleteAccount deletes the account with the given name, using the password to
+// ensure access. If the account doesn't exist, then return ErrCryptoKeyNotFound.
+// If the password is wrong, then return ErrDecryptionFailed.
+func (s *gnomobileService) DeleteAccount(ctx context.Context, req *connect.Request[rpc.DeleteAccountRequest]) (*connect.Response[rpc.DeleteAccountResponse], error) {
+	if err := s.getSigner().Keybase.Delete(req.Msg.NameOrBech32, req.Msg.Password, false); err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			return nil, rpc.ErrCode_ErrCryptoKeyNotFound
+		} else if strings.Contains(err.Error(), "ciphertext decryption failed") {
```
Comment (D4ryl00) [inline service/api.go:166]: It's better to compare against a const error message exported from Gno. Is it possible?

Comment (jefft0) [reply service/api.go]: OK. I pushed a comment to check the error type.
https://github.com/gnolang/gnomobile/pull/43/commits/c309af58475da3aefb13b1c0a81aad1bb34872b1

**Review by D4ryl00 — APPROVED** [review]
LGTM :+1:

**Pull request merged**